### ISSUE TITLE
Iterating over all Properties

### ISSUE THREAD

**Issue by furesoft** [MrKWatkins/Ast#27]
For a printer I need to iterate over all properties for nodes that I don't know. I am using reflection but using the Properties property would be faster. Can you make `Properties` public instead of protected?

**Comment by MrKWatkins** [comment]
Let me have a think about this one. I chose to keep Properties protected so it couldn't be accessed from outside as raw access would be untyped and people could do unexpected things. But I can see iterating over them would be useful. Maybe just access to read only iteration without allowing properties to be mutated would be fine? Note that you're not guaranteed to have a value for every property - only keys that have been explicitly set are kept.

**Comment by furesoft** [comment]
A readonly view would be great

**Comment by MrKWatkins** [comment]
I've added an `EnumerateProperties` method to `PropertyNode` that will give you `KeyValuePair<string, object>`s for the properties. If it's a single valued property Value will just contain the value, if it's a multiple valued property Value will be a `List<object>` containing the values.

I've published a new package with the change, https://www.nuget.org/packages/MrKWatkins.Ast/0.9.124.

Is that good enough?
